src/main/java/com/example/demo/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.example.demo.repository;
+
+import com.example.demo.model.Book;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private SessionFactory sessionFactory;
+
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't save data", e);
```

Comment:
message should be more informative 
```suggestion
            throw new DataProcessingException("Can't save data", e);
```